conans/test/integration/build_requires/build_requires_test.py

```diff
@@ -628,3 +628,44 @@ def test_track_host_error_wrong_context(self, build_profile):
         c.save({"conanfile.py": GenConanfile("pkg").with_requirement("protobuf/<host_version>")})
         c.run(f"install . {build_profile}", assert_error=True)
         assert "uses '<host_version>' in requires" in c.out
+
+    @pytest.mark.parametrize("build_profile", [False, True])
+    def test_host_version_test_package(self, build_profile):
+        """
+        https://github.com/conan-io/conan/issues/14704
+        """
+        c = TestClient()
+        pkg = textwrap.dedent("""
+                from conan import ConanFile
+                class ProtoBuf(ConanFile):
+                    name = "pkg"
+                    version = "0.1"
+                    def requirements(self):
+                        self.requires("protobuf/[>=1.0]")
+                    def build_requirements(self):
+                        self.tool_requires("protobuf/<host_version>")
+                """)
+        # regular requires test_package
+        c.save({"protobuf/conanfile.py": GenConanfile("protobuf"),
+                "pkg/conanfile.py": pkg,
+                "pkg/test_package/conanfile.py": GenConanfile().with_test("pass")})
+        c.run("create protobuf 1.0@ -s compiler.version=14 -s:b compiler.version=14")
```

Comment:
```suggestion
        c.run("create protobuf 1.0@ -pr:b=default")
```
Enough to trigger the 2 profiles failure?

Reply:
Ahhhg. I had to modify the test for this to run locally on my end, but forgot to remove the change before pushing